app/code/Magento/Sales/Model/Order/StateResolver.php

```diff
@@ -22,7 +22,7 @@ class StateResolver implements OrderStateResolverInterface
     private function isOrderComplete(OrderInterface $order)
     {
         /** @var $order Order|OrderInterface */
-        if (0 == $order->getBaseGrandTotal() || $order->canCreditmemo()) {
+        if ((!$order->canCreditmemoForZeroAll() && 0 == $order->getBaseGrandTotal()) || $order->canCreditmemo()) {
```

Comment:
I am not sure about the `0 == $order->getBaseGrandTotal()` that's why I didn't remove it.
@orlangur can you suggest what should we do?

Reply:
Hi @sanjaychouhan-webkul,

Can we just remove this line and check if Integration tests will fail?
Probably, this is only 'hack' was done for zero-total orders.

It seems really strange, because it is difficult to imagine situation where `0 == $order->getBaseGrandTotal()` will be `false` and 
https://github.com/magento/magento2/blob/211dd25c73bb2539c3c064374d9d5649f253fd14/app/code/Magento/Sales/Model/Order.php#L667
will be `true`.

Reply:
I have removed it in my last commit and Integration Test was successfull. Please check. @swnsma 
https://github.com/magento/magento2/blob/ff2a9fbc7c300ab6257c27f8b84a89fb35cd11da/app/code/Magento/Sales/Model/Order/StateResolver.php#L22-L25